apps/s_client.c

```diff
@@ -1742,7 +1762,11 @@ int s_client_main(int argc, char **argv)
     if (fallback_scsv)
         SSL_set_mode(con, SSL_MODE_SEND_FALLBACK_SCSV);
 
-    if (servername != NULL) {
+    if (!noservername && (servername != NULL || dane_tlsa_domain == NULL)) {
+        if (servername == NULL)
+            servername = host;
+        if (servername == NULL)
+            servername = "localhost";
```

Comment:
> Probably better to not bother sending SNI if you can't come up with one.

If `host` is NULL then ultimately we call `getaddrinfo` with the first argument (`node`) as NULL. According to the (Linux) documentation it says:

````
       If node is NULL, then the network address will  be  set
       to  the loopback interface address (INADDR_LOOPBACK for IPv4 addresses,
       IN6ADDR_LOOPBACK_INIT for IPv6 address); this is used  by  applications
       that intend to communicate with peers running on the same host.
````

So it seems reasonable to me to use `localhost` here??

> That perhaps makes some sense, but will likely involve further changes to the tests and docs.

I did have to make some changes to the tests for this, but they are already included in this PR. I also mentioned it in the s_client docs. Not sure if you envisaged anything else beyond those things.